### PR TITLE
feat: Christmas Theme

### PR DESCRIPTION
**Pull request by akanex115**
# Background
https://github.com/ajktown/ConsistencyGPT/issues/71

## Checklist Before PR Review
- [x] The following has been handled:
  -  `Draft` is set for this PR
  - `Title` is checked
  - `Background` is filled
  - `Assignee` is set
  - `Labels` are set
  - `development` is linked if related issue exists

## Checklist (Right Before PR Review Request)
- [x] The following has been handled:
  - Final Operation Check is done
  - Mobile View Operation Check is done
  - Make this PR as an open PR

## Checklist (Reviewers)
- [x] Check if there are any other missing TODOs that are not yet listed
- [x] Review Code
- [x] Every item on the checklist has been addressed accordingly
- [x] If `development` is associated to this PR, you must check if every TODOs are handled

### CODE REVIEW

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 1]
```diff
@@ -9,12 +9,22 @@ enum Basic {
...
-  'three' = `#ffc500`,
-  'four' = `#fe9601`,
-  'five' = `#d47901`,
+// enum Halloween {
+//   'one' = `#ebedf0`,
+//   'two' = `#ffee4a`,
+//   'three' = `#ffc500`,
+//   'four' = `#fe9601`,
+//   'five' = `#d47901`,
+// }
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]: don't comment it out!

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 2]
```diff
@@ -9,12 +9,22 @@ enum Basic {
...
+
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts:19]: Did you run `yarn inspect`?

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 3]
```diff
@@ -9,12 +9,22 @@ enum Basic {
...
+enum Christmas {
+  'one' = `#edf8e9`,
+  'two' = `#c7e9c0`,
+  'three' = `#74c476`,
+  'four' = `#238b45`,
+  'five' = `#005a32`,
+  'six' = `#b01b2e`,
+  'seven' = `#ffffff`,
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]: Only five colors please xD (just like `Halloween` or `Basic`

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 4]
```diff
@@ -25,20 +35,37 @@ export const getAppThemeColorLambda = (theme?: AppTheme) => {
...
+  //     Halloween.four,
+  //   ],
+  //   dark: [
+  //     Halloween.one,
+  //     Halloween.two,
+  //     Halloween.three,
+  //     Halloween.four,
+  //     Halloween.four,
+  //   ],
+  // }
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]: Don't comment out

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 5]
```diff
@@ -25,20 +35,37 @@ export const getAppThemeColorLambda = (theme?: AppTheme) => {
...
+
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]: Use something like
`if (theme === AppTheme.Christmas) { ... }`

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 6]
```diff
@@ -57,17 +85,31 @@ export const getButtonColorLambda = (
...
 
   // by default halloween as only two options for now:
+  // if (disabled) {
+  //   return {
+  //     color: Halloween.three, // it will be the font color
+  //     borderColor: Halloween.one, // it will be the border color
+  //     backgroundColor: Halloween.one,
+  //   }
+  // }
+
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]: Don't comment it out!

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 7]
```diff
@@ -57,17 +85,31 @@ export const getButtonColorLambda = (
...
+      backgroundColor: Christmas.seven,
     }
   }
 
+  // return {
+  //   color: disabled ? Halloween.three : Halloween.five,
+  //   borderColor: disabled ? Halloween.one : Halloween.five,
+  //   backgroundColor: disabled ? undefined : Halloween.two,
+  // }
+
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]: if (theme === AppTheme.Christmas) { ... }

File: src/recoil/app-theme/app-theme.state.ts [hunk 8]
```diff
@@ -19,22 +20,23 @@ const isWithinRange = (date: DateTime<true> | DateTime<false>) => {
...
+const christmas = DateTime.fromObject({ month: 12, day: 25 }).endOf(`day`)
 const defaultTheme = ((): AppTheme => {
   // TODO: The following themes will be implemented in the future:
   // if (isWithinRange(newYears)) return AppTheme.NewYear // Happy New Year theme
   // if (isWithinRange(valentines)) return AppTheme.Valentines // Love theme
   // if (isWithinRange(spring)) return AppTheme.Spring // Cherry blossom theme
   // if (isWithinRange(summer)) return AppTheme.Summer // Beach theme
-  if (isWithinRange(halloween)) return AppTheme.Halloween
-  // if (isWithinRange(christmas)) return AppTheme.Chrstimas // Christmas theme
+  // if (isWithinRange(halloween)) return AppTheme.Halloween
```
Comment (mlajkim) [inline src/recoil/app-theme/app-theme.state.ts]: dont comment it out!
```suggestion
  if (isWithinRange(halloween)) return AppTheme.Halloween
```

File: src/recoil/app-theme/app-theme.state.ts [hunk 9]
```diff
@@ -19,22 +20,23 @@ const isWithinRange = (date: DateTime<true> | DateTime<false>) => {
...
   // Default:
   return AppTheme.Basic
 })()
 
 export const appThemeState = atom<AppTheme>({
   key: Rkp.AppTheme,
-  default: defaultTheme,
-  // default: AppTheme.Halloween, // You can always override the default theme here, and compile will fail if you forget to remove it
+  //default: defaultTheme,
+  default: AppTheme.Christmas, // You can always override the default theme here, and compile will fail if you forget to remove it
```
Comment (mlajkim) [inline src/recoil/app-theme/app-theme.state.ts]: Back to original setting before PR please!
```suggestion
  default: defaultTheme,
  // default: AppTheme.Christmas, // You can always override the default theme here, and compile will fail if you forget to remove it
```

**Review by mlajkim — COMMENTED** [review]
first review

File: src/recoil/app-theme/app-theme.state.ts [hunk 10]
```diff
@@ -19,15 +20,16 @@ const isWithinRange = (date: DateTime<true> | DateTime<false>) => {
   return now >= date.minus({ days: 13 }) && now <= date.plus({ days: 1 })
 }
 
-const halloween = DateTime.fromObject({ month: 10, day: 31 }).endOf(`day`)
+//const halloween = DateTime.fromObject({ month: 10, day: 31 }).endOf(`day`)
```
Comment (mlajkim) [inline src/recoil/app-theme/app-theme.state.ts]: dont comment this out either!

**Review by mlajkim — COMMENTED** [review]
review 2

File: src/lambdas/get-app-theme-color.lambda.ts (hunk 7 above)
Comment (akanex115) [reply src/lambdas/get-app-theme-color.lambda.ts]: Handled in [8ef685e](https://github.com/ajktown/ConsistencyGPT/pull/74/commits/8ef685e54a10fbb3a6fff4f9964b248e91c083eb)

File: src/lambdas/get-app-theme-color.lambda.ts (hunk 1 above)
Comment (akanex115) [reply src/lambdas/get-app-theme-color.lambda.ts]: Handled in [8ef685e](https://github.com/ajktown/ConsistencyGPT/pull/74/commits/8ef685e54a10fbb3a6fff4f9964b248e91c083eb)

File: src/lambdas/get-app-theme-color.lambda.ts (hunk 2 above)
Comment (akanex115) [reply src/lambdas/get-app-theme-color.lambda.ts:19]: yes
image

File: src/lambdas/get-app-theme-color.lambda.ts (hunk 4 above)
Comment (akanex115) [reply src/lambdas/get-app-theme-color.lambda.ts]: Handled in [8ef685e](https://github.com/ajktown/ConsistencyGPT/pull/74/commits/8ef685e54a10fbb3a6fff4f9964b248e91c083eb)

File: src/lambdas/get-app-theme-color.lambda.ts (hunk 5 above)
Comment (akanex115) [reply src/lambdas/get-app-theme-color.lambda.ts]: Handled in [8ef685e](https://github.com/ajktown/ConsistencyGPT/pull/74/commits/8ef685e54a10fbb3a6fff4f9964b248e91c083eb)

File: src/lambdas/get-app-theme-color.lambda.ts (hunk 6 above)
Comment (akanex115) [reply src/lambdas/get-app-theme-color.lambda.ts]: Handled in [8ef685e](https://github.com/ajktown/ConsistencyGPT/pull/74/commits/8ef685e54a10fbb3a6fff4f9964b248e91c083eb)

File: src/lambdas/get-app-theme-color.lambda.ts (hunk 7 above)
Comment (akanex115) [reply src/lambdas/get-app-theme-color.lambda.ts]: Handled in [8ef685e](https://github.com/ajktown/ConsistencyGPT/pull/74/commits/8ef685e54a10fbb3a6fff4f9964b248e91c083eb)

File: src/recoil/app-theme/app-theme.state.ts (hunk 10 above)
Comment (akanex115) [reply src/recoil/app-theme/app-theme.state.ts]: Handled in [8ef685e](https://github.com/ajktown/ConsistencyGPT/pull/74/commits/8ef685e54a10fbb3a6fff4f9964b248e91c083eb)

File: src/lambdas/get-app-theme-color.lambda.ts (hunk 3 above)
Comment (akanex115) [reply src/lambdas/get-app-theme-color.lambda.ts]: How about this
image

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 11]
```diff
@@ -25,29 +33,49 @@ export const getAppThemeColorLambda = (theme?: AppTheme) => {
     }
 
   // by default halloween as only two options for now:
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts:35]: This should be somewhere

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 12]
```diff
@@ -25,29 +33,49 @@ export const getAppThemeColorLambda = (theme?: AppTheme) => {
...
+    }
 }
 
 export const getAppBarColorLambda = (theme?: AppTheme) => {
   if (!theme || theme === AppTheme.Basic) return undefined // use the default color
 
   // by default halloween as only two options for now:
-  return Halloween.four
+  //return Halloween.four
+  return Christmas.five
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]:
```suggestion
　if (theme === AppTheme.Halloween) return Halloween.four
  return Christmas.five
```

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 13]
```diff
@@ -25,29 +33,49 @@ export const getAppThemeColorLambda = (theme?: AppTheme) => {
...
+      dark: [
+        Halloween.one,
+        Halloween.two,
+        Halloween.three,
+        Halloween.four,
+        Halloween.four,
+      ],
+    }
   }
+  if (theme === AppTheme.Christmas)
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]: Why do we need the return at the end?

File: src/lambdas/get-app-theme-color.lambda.ts [hunk 14]
```diff
@@ -59,15 +87,30 @@ export const getButtonColorLambda = (
   // by default halloween as only two options for now:
   if (disabled) {
     return {
-      color: Halloween.three, // it will be the font color
+      color: Halloween.one, // it will be the font color
       borderColor: Halloween.one, // it will be the border color
-      backgroundColor: Halloween.one,
+      backgroundColor: Halloween.five,
```
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]: dont modify halloween color for Christmas pr

File: src/recoil/app-theme/app-theme.state.ts [hunk 15]
```diff
@@ -20,21 +21,22 @@ const isWithinRange = (date: DateTime<true> | DateTime<false>) => {
...
   // Default:
   return AppTheme.Basic
 })()
 
 export const appThemeState = atom<AppTheme>({
   key: Rkp.AppTheme,
-  default: defaultTheme,
-  // default: AppTheme.Halloween, // You can always override the default theme here, and compile will fail if you forget to remove it
+  //default: defaultTheme,
+  default: AppTheme.Christmas, // You can always override the default theme here, and compile will fail if you forget to remove it
```
Comment (mlajkim) [inline src/recoil/app-theme/app-theme.state.ts]:
```suggestion
  default: defaultTheme,
  // default: AppTheme.Christmas, // You can always override the default theme here, and compile will fail if you forget to remove it
```

File: src/pages/users/[nickname].tsx [hunk 16]
```diff
@@ -5,7 +5,7 @@ import RitualsFrame from '@/components/organism_rituals_frame'
 import Appbar from '@/components/organism_appbar'
 import StyledCentered from '@/organisms/StyledCentered'
 import ThemedTextButtonAtom from '@/atoms_themed/ThemedTextButton'
-import { Grid2, Stack, Typography } from '@mui/material'
+import { Grid, Stack, Typography } from '@mui/material'
```
Comment (mlajkim) [inline src/pages/users/[nickname].tsx]: Dont change this file (Use `Grid2`)

**Review by mlajkim — COMMENTED** [review]
code review

File: src/pages/users/[nickname].tsx (hunk 16 above)
Comment (akanex115) [reply src/pages/users/[nickname].tsx]: Handled in [9c2c78d](https://github.com/ajktown/ConsistencyGPT/pull/74/commits/9c2c78d9c3a34a0208dd1a971f2e1a692733a9d9)

File: src/pages/users/[nickname].tsx [hunk 17]
```diff
@@ -5,9 +5,10 @@ import RitualsFrame from '@/components/organism_rituals_frame'
 import Appbar from '@/components/organism_appbar'
 import StyledCentered from '@/organisms/StyledCentered'
 import ThemedTextButtonAtom from '@/atoms_themed/ThemedTextButton'
-import { Grid2, Stack, Typography } from '@mui/material'
+import { Stack, Typography } from '@mui/material'
 import { useUserByNickname } from '@/hooks/user/get-user-by-nickname.hook'
 import StyledUserAvatar from '@/atoms/StyledUserAvatar'
+import Grid2 from '@mui/material/Unstable_Grid2'
```
Comment (mlajkim) [inline src/pages/users/[nickname].tsx]: Dont make changes!: 
```suggestion
import { Grid2, Stack, Typography } from '@mui/material'
import { useUserByNickname } from '@/hooks/user/get-user-by-nickname.hook'
import StyledUserAvatar from '@/atoms/StyledUserAvatar'
```

File: src/lambdas/get-app-theme-color.lambda.ts (hunk 13 above)
Comment (mlajkim) [inline src/lambdas/get-app-theme-color.lambda.ts]: No If!
```suggestion
```

**Review by mlajkim — COMMENTED** [review]
review

**Review by mlajkim — COMMENTED** [review]
Did you run `yarn inspect`?

**Review by mlajkim — APPROVED** [review]
Lgtm

**Pull request merged**